DQM/L1TMonitor/src/L1TdeStage2CaloLayer2.cc

```diff
@@ -0,0 +1,1234 @@
+  */
+
+  TH1F * agreementHist = agreementSummary->getTH1F();
+
+  l1t::EGammaBxCollection::const_iterator dataEGIt = egDataCol->begin(currBx);
+  l1t::EGammaBxCollection::const_iterator emulEGIt = egEmulCol->begin(currBx);
+
+  l1t::TauBxCollection::const_iterator dataTauIt = tauDataCol->begin(currBx);
+  l1t::TauBxCollection::const_iterator emulTauIt = tauEmulCol->begin(currBx);
+
+  l1t::EtSumBxCollection::const_iterator dataSumIt = sumDataCol->begin(currBx);
+  l1t::EtSumBxCollection::const_iterator emulSumIt = sumEmulCol->begin(currBx);
+
+  if (!compareJets(jetDataCol, jetEmulCol)) {
+    eventGood = false;
+  }
+
+  if (!compareEGs(egDataCol, egEmulCol)) {
+    eventGood = false;
+  }
+
+  if (!compareTaus(tauDataCol, tauEmulCol)) {
+    eventGood = false;
+  }
+
+  if (!compareSums(sumDataCol, sumEmulCol)) {
+    eventGood = false;
+  }
+
+  /**
+     Questions:
+     - what could make the data and emul bx ranges to be different?
+     - how can I confirm that the emulator data is being filled?
+  */
+
+  if (eventGood) {
+    agreementHist->Fill(EVENTGOOD);
+  }
+
+  agreementHist->Fill(NEVENTS);
+  problemSummary->getTH1F()->Fill(NEVENTS_P);
+}
+
+// comparison method for jets
+bool L1TdeStage2CaloLayer2::compareJets(
+  const edm::Handle<l1t::JetBxCollection> & dataCol,
+  const edm::Handle<l1t::JetBxCollection> & emulCol)
+{
+  bool eventGood = true;
+
+  l1t::JetBxCollection::const_iterator dataIt = dataCol->begin(currBx);
+  l1t::JetBxCollection::const_iterator emulIt = emulCol->begin(currBx);
+
+  TH1F * objEtHistData = jetEtData->getTH1F();
+  TH1F * objPhiHistData = jetPhiData->getTH1F();
+  TH1F * objEtaHistData = jetEtaData->getTH1F();
+
+  TH1F * objEtHistEmul = jetEtEmul->getTH1F();
+  TH1F * objPhiHistEmul = jetPhiEmul->getTH1F();
+  TH1F * objEtaHistEmul = jetEtaEmul->getTH1F();
+
+  TH1F * objSummaryHist = jetSummary->getTH1F();
+  TH1F * summaryHist = agreementSummary->getTH1F();
+
+  // process jets
+  if (dataCol->size() != emulCol->size()) {
+
+    if (dataCol->size() < emulCol->size()) {
+      if (dataCol->size() < 1) {
+	return false;
+      }
+
+      while (true) {
+	objEtHistData->Fill(dataIt->hwPt());
+	objEtaHistData->Fill(dataIt->hwEta());
+	objPhiHistData->Fill(dataIt->hwPhi());
+
+	++dataIt;
+
+	if (dataIt == dataCol->end(currBx))
+	  break;
+      }
+    } else {
+
+      if (emulCol->size() < 1) {
+	return false;
+      }
+
+      while (true) {
+
+	objEtHistEmul->Fill(emulIt->hwPt());
+	objEtaHistEmul->Fill(emulIt->hwEta());
+	objPhiHistEmul->Fill(emulIt->hwPhi());
+
+	++emulIt;
+
+	if (emulIt == emulCol->end(currBx))
+	  break;
+      }
+    }
+
+    problemSummary->getTH1F()->Fill(JETCOLLSIZE);
+    return false;
+  }
+
+  if (dataIt != dataCol->end(currBx) ||
+      emulIt != emulCol->end(currBx)) {
+    while(true) {
+
+      bool posGood = true;
+      bool etGood = true;
+
+      // jet Et mismatch
+      if (dataIt->hwPt() != emulIt->hwPt()) {
+        etGood = false;
+ 	eventGood = false;
```

Comment:
please fix the indentation here

Reply:
Hi @davidlange6, both indentation and comment in this location have been updated. To confirm, please have a look at the whole file individually. Other comments are welcome.

Thanks